### DISCUSSION TITLE
Notch's own dimensions

### DISCUSSION THREAD

**Discussion by toysonlocation** [airsdk/Adobe-Runtime-Support#2090] [Q&A]
Is there any mechanism to get the notch's actual dimensions? I have a need to know what space it actually occupies.

Thanks!

**Comment by ajwfrost** [comment]
Currently via an ANE, I believe @marchbold  - but we have got this capability in our next version of AIR which we'll beta soon. It's not any exact dimensions but instead gives a 'safe area' so that you can tell e.g. how much vertical space may be taken up at the top of the screen by a notch. Afaik there's no way we can find out how wide a notch might be in case you wanted to know how much space is available in this unsafe area either side of the physical cut-out, but there may be information you can add manually based on phone model detection?

**Reply by toysonlocation** [reply]
Yes, I do need the notch itself so I know the "shape" of the usable unsafe area. I am hoping to avoid a device table. Do you know if this possible natively? As in, a candidate for an ANE?

**Reply by ajwfrost** [reply]
I am wondering whether it's possible on Android, e.g.
https://developer.android.com/reference/android/view/WindowManager#getCurrentWindowMetrics()
https://developer.android.com/reference/android/view/WindowMetrics#getWindowInsets()
https://developer.android.com/reference/android/view/WindowInsets#getInsets(int) - using `DISPLAY_CUTOUT`

on iOS: I'm not sure it's possible.... e.g.
https://developer.apple.com/documentation/uikit/uiview/2891103-safeareainsets?language=objc
but that just gives an inset left/right/top/bottom figure (and I suspect is what we're using..)

thanks

**Reply by toysonlocation** [reply]
@ajwfrost Thanks for digging into it, Andrew. iOS is actually my immediate concern, so it looks out-of-bounds. I will keep my fingers crossed that next time Apple errs and shares.

**Reply by marchbold** [reply]
Hi, yes our Application ANE will give you this information and our next release will have the additional information on Android, (just in testing now). As far as I'm aware iOS has no more detailed information, however the devices are much more consistent so it is "possible though intensive" to estimate it based on the device.

**Reply by marchbold** [reply]
Documentation on it is here: https://docs.airnativeextensions.com/docs/application/display/cutouts
